src/highlight/highlightUtil.ts

```diff
@@ -94,17 +94,37 @@ export const dedupeRects = (rects: Shape[], threshold = 0.1): Shape[] => {
 };
 
 export const combineRectsByRow = (rects: Shape[]): Shape[] => {
-    const dedupedRects = dedupeRects(rects);
-
-    const rows = dedupedRects.reduce((groups, rect) => {
-        const { y } = rect;
+    const rows = dedupeRects(rects).reduce((groups, rect) => {
+        const { height, width, x, y } = rect;
         const roundedY = Math.round(y);
-        groups[roundedY] = groups[roundedY] || [];
-        groups[roundedY].push(rect);
+        const keys = Object.keys(groups).map(key => parseInt(key, 10));
+        const lastYKey = keys.length ? keys[keys.length - 1] : null;
```

Comment:
Should we use an array instead of a hash table if we only care about the last value?

Reply:
Good idea 💡 , let me give that a try